ckanext/canada/logic.py

```diff
@@ -185,3 +194,88 @@ def datastore_create_temp_user_table(context):
             username=literal_string(username),
             sysadmin='TRUE' if is_sysadmin(username) else 'FALSE'))
 
+
+@chained_action
+def canada_resource_view_show(up_func, context, data_dict):
+    """
+    Return the metadata of a resource_view.
+
+    :param id: the id of the resource_view
+    :type id: string
+
+    :rtype: dictionary
+
+    Canada Fork: extends the resource_view_show action method
+    to prevent showing datatable_views for invalid and inactive Resources
+
+    We need this method to 404 the resource_view page (e.g. viewing in full screen)
+
+    If a user is logged in, we want them to be able to see and edit the datatables_view
+    views still. We will just add a key to the view dict to be used within templates for visuals.
+    """
+    view_dict = up_func(context, data_dict)
+    if view_dict.get('view_type') == 'datatables_view':
+        # at this point, the core function has been called, calling resource_show
+        # and adding the Resource Object into the context
+        res_dict = context.get('resource').__dict__
```

Comment:
this will not a normal way to pass resource data, either use the resource object or the dictized version from resource_show/package_show not the resource object's `__dict__`